### ISSUE TITLE
Downloaded Artifact Verification at Peer/Agent Node

### ISSUE THREAD

**Issue by dasmanas** [pyrsia/pyrsia#1533]

**Comment by karolh2000** [comment]
@dasmanas I think this issue can be split into two tasks. 
1) The first one is about creating the verification mechanism in the node logic (node and authorized node) + REST API.
2) Task 2 (once the task 1 is merged) - implement the client side which uses the REST API.

**Comment by dasmanas** [comment]
### Tasks to solve
- [ ] Docker disk file location in macOS `~/Library/Containers/com.docker.docker/Data/vms/0/data/Docker.raw`. Finding an image from raw and calculate the the SHA is an uphill task.
- [ ] `docker inspect --format='id={{ json .Id }}, layers={{ json .RootFS.Layers }}' <image>:<tag>` to fetch id and layers information.

**Comment by dasmanas** [comment]
> @dasmanas I think this issue can be splitted into two tasks.
> 
> 1. The first one is about creating the verification mechanism in the node logic (node and authorized node) + REST API.
> 2. Task 2 (once task 1 is merged) - implement the client side which uses the REST API.

We already have another issue #1302 for the server/node side verification.

**Comment by dasmanas** [comment]
depends on #1557
